roles/map-role/tasks/main.yml

```diff
@@ -134,7 +134,7 @@
 
 - name: "Assigning Role - {{rolename}}"
   uri:
-    url: "https://{{du_fqdn}}/resmgr/v1/hosts/{{host_id.stdout.strip()}}/roles/{{rolename}}"
+    url: "https://{{du_fqdn}}/resmgr/v1/hosts/{{host_id.stdout.strip()}}/roles/{{rolename}}/versions/{{version}}"
```

Comment:
@raghuP9 - this will break PMO related operations since the same role is reused there. We could replicate the role and make it PMK specific

Reply:
I can add another step which is duplicate of this one and make it optional if roleversion is provided and do the opposite in the original step.